os/drivers/audio/syu645b.c

```diff
@@ -1031,15 +1041,187 @@ static void syu645b_hw_reset_config(FAR struct syu645b_dev_s *priv)
+		meddbg("Failed to allocate memory to hold equalizer json file content.\n");
+		return ERROR;
+	}
+	int fd = open(PROD_EQUALIZE_ENCODE_JSON_PATH, O_RDONLY);
+	if (fd == -1) {
+		meddbg("Failed to open equalizer json.\n");
+		ret = ERROR;
+		goto err_with_buffer;
+	}
+	ssize_t bytesRead = read(fd, buffer, st.st_size);
+	if (bytesRead == -1 || bytesRead != st.st_size) {
+		meddbg("Failed to read from equalizer json.\n");
+		ret = ERROR;
+		goto err_with_fd;
+	}
+	
+	gJSON = cJSON_Parse(buffer);
+	
+	if (!gJSON) {
+		meddbg("Failed to parse equalizer json file.\n");
+		ret = ERROR;
+		goto err_with_fd;
+	}
+
+err_with_fd:
+	close(fd);
+err_with_buffer:
+	close(buffer);
+	return ret;
+}
+
+static int dec_to_hex_string(uint32_t decimal, char *hexadecimal){
+	if (decimal > EQUALIZER_PRESET_MAX) {
+		meddbg("Input invalid preset.\n");
+		return ERROR;
+	}
+	
+	int pos = 9;
+	hexadecimal[0] = '0';
+	hexadecimal[1] = 'x';
+	hexadecimal[10] = '\0';
+	while (1) {
+		int mod = decimal % 16;
+		if (mod < 10) {
+			hexadecimal[pos] = 48 + mod;
+		}
+		else {
+			hexadecimal[pos] = 65 + (mod - 10);
+		}
+		decimal = decimal / 16;
+		pos--;
+
+		if (decimal == 0) {
+			if (pos > 1) {
+				for(int i=2; i<=pos; i++) {
+				hexadecimal[pos] = '0';
+				}
+			}
+			break;
+		}
+	}
+
+	return OK;
+}
+
+static int string_to_script(char *string, t_codec_init_script_entry *script){
+	int cnt = 0;
+
+	const char *ptr = string;
+	while (*ptr && cnt < SYU645B_REG_DATA_TYPE_MAX) {
+		if (*ptr == '0' && (*(ptr + 1) == 'x' || *(ptr + 1) == 'X')) {
+			char *end_ptr;
+			uint8_t val = strtol(ptr, &end_ptr, 16);
+
+			if (ptr != end_ptr) {
+				if (cnt == 0) {
+					script->addr = val;
+				}
+				else {
+					script->val[cnt-1] = val;	
+				}
+				ptr = end_ptr;
+				cnt++;
+				continue;
+			}
+		}
+		ptr++;
+	}
+
+	script->delay = 0;
+	script->type = cnt;
+
+	return OK;
+}
+
+static int syu645b_set_basic_equalizer(FAR struct syu645b_dev_s *priv){
+	syu645b_exec_i2c_script(priv, t_codec_dq_preset_default_script, sizeof(t_codec_dq_preset_default_script) / sizeof(t_codec_init_script_entry));
+	priv->max_volume = 0x9F;
+
+	/* And then setVolume Again */
+	if (!priv->mute) {
+		syu645b_setvolume(priv);
+	}
+}
+
 static void syu645b_set_equalizer(FAR struct syu645b_dev_s *priv, uint32_t preset)
 {
 	/* Reset First */
 	syu645b_reset_config(priv);
 
-	/* And then Set default EQ Set */
-	syu645b_exec_i2c_script(priv, t_codec_dq_preset_default_script, sizeof(t_codec_dq_preset_default_script) / sizeof(t_codec_init_script_entry));
-	priv->max_volume = 0x9F;
+	/* Initialize for i2c write */
+	uint8_t reg[SYU645B_REG_DATA_TYPE_MAX];
+	FAR struct i2c_dev_s *dev = priv->i2c;
+	FAR struct i2c_config_s *syu645b_i2c_config = &(priv->lower->i2c_config);
+	
+	int ret = 0;
+	char c_preset[EQUALIZER_PRESET_LEN];
 
+	/* Convert preset hex to string for key & Parse equlizer json to gJSON */
+	if (dec_to_hex_string(preset, c_preset) == ERROR) {
+		syu645b_set_basic_equalizer(priv);
+		return;
```

Comment:
I think syu645b_set_equalizer need to be changed to int.
here it should return -EINVAL (because preset is not valid)
hmm and I think we already set basic eq value in syu645b_initialize, syu645b_set_basic_equalizer can be removed.